Align `selectorText` to not serialize `@page` and not be nullable  as per Web Specification

Ahmad-S792 opened the pull request:
#### 4e7e8cdf99fa7ad7c2779f5509d74f9d9fc9fc10
<pre>
Align `selectorText` to not serialize `@page` and not be nullable  as per Web Specification

<a href="https://bugs.webkit.org/show_bug.cgi?id=271432">https://bugs.webkit.org/show_bug.cgi?id=271432</a>
<a href="https://rdar.apple.com/125588212">rdar://125588212</a>

Reviewed by Darin Adler.

This patch aligns WebKit with Web Specification [1] &amp; [2]:

[1] <a href="https://drafts.csswg.org/cssom/#dom-csspagerule-selectortext">https://drafts.csswg.org/cssom/#dom-csspagerule-selectortext</a>
[2] <a href="https://drafts.csswg.org/cssom/#csspagerule">https://drafts.csswg.org/cssom/#csspagerule</a>

As per Web Specification, it should not append `@page`, so now
it serialize correctly. Similarly, it should not be nullable.

This patch also syncs WPT tests (not printing specific) below:

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/c72ee0c082d350b8b44f7a6bac1f028752b13527">https://github.com/web-platform-tests/wpt/commit/c72ee0c082d350b8b44f7a6bac1f028752b13527</a>

* Source/WebCore/css/CSSPageRule.cpp:
(WebCore::CSSPageRule::selectorText const):
(WebCore::CSSPageRule::cssText const):
* Source/WebCore/css/CSSPageRule.idl:

&gt; Rebaselines (Local Tests)
* LayoutTests/http/wpt/cssom/CSSPageRule.html: Updated
* LayoutTests/fast/dom/css-element-attribute-js-null-expected.txt: Synced from Blink Source
* LayoutTests/fast/dom/css-element-attribute-js-null.html: Synced from Blink Source

&gt; WPT Tests Update &amp; Sync:
* LayoutTests/imported/w3c/resources/import-expectations.json:
* * LayoutTests/imported/w3c/web-platform-tests/css/cssom/cssom-pagerule.html: Extended for `cssText` cases as well
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/cssom-pagerule-expected.txt: Rebaselined
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/margin-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/margin-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/margin-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/margin-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/margin-003-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/margin-003.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/page-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/page-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/page-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/page-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/inheritance-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/inheritance.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/META.yml:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-orientation.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-orientation.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-000-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-000.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-003-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-003.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-004-expected.txt::
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-004.html::
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/margin-rules-001-expected.txt::
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/margin-rules-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/margin-rules-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/margin-rules-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/nested-rules-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/nested-rules-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-invalid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-invalid.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-orientation-computed.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-orientation-computed.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-orientation-invalid.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-orientation-invalid.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-rules-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-rules-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-valid.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/WEB_FEATURES.yml:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/trailing-declaration-crash.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/WEB_FEATURES.yml:

Canonical link: <a href="https://commits.webkit.org/289428@main">https://commits.webkit.org/289428@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3a2ffba9d16682ba1fee228c355b8598b66f2f2f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/86868 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/6375 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/41210 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/91716 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/37600 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/88917 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/6643 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/14438 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/67150 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/24925 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/89871 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/5060 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/78625 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/47469 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/4845 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/33005 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/36718 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/75350 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/33867 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/93609 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/14021 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/10181 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/75950 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/14222 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/74470 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/75146 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18484 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/19469 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/17882 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/6854 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/14044 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/19305 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/13782 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/17227 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/15567 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Ahmad-S792 commented on LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/page-001-expected.txt:
```diff
@@ -0,0 +1,6 @@
+
```
Progressed.

Ahmad-S792 commented on LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-orientation.tentative-expected.txt:
```diff
@@ -0,0 +1,13 @@
+FAIL contents for selector [':right'] assert_equals: unexpected @page contents expected "page-orientation: rotate-right;" but got ""
+FAIL contents for selector [':first'] assert_equals: unexpected @page contents expected "page-orientation: rotate-left;" but got ""
+FAIL contents for selector ['named1'] assert_equals: unexpected @page contents expected "page-orientation: upright;" but got ""
+PASS contents for selector ['named2']
```
Progressed.

Ahmad-S792 commented on LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-000-expected.txt:
```diff
@@ -0,0 +1,23 @@
+
```
Progressed.

Ahmad-S792 commented on LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-001-expected.txt:
```diff
@@ -0,0 +1,7 @@
+
```
Progressed.

Ahmad-S792 commented on LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-003-expected.txt:
```diff
@@ -0,0 +1,6 @@
+
```
Partially Progressed

Ahmad-S792 commented on LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-004-expected.txt:
```diff
@@ -0,0 +1,10 @@
+
```
Partially progressed.

Ahmad-S792 commented on LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-rules-001-expected.txt:
```diff
@@ -0,0 +1,10 @@
+
```
Regression - need to investigate - hence WIP.

Ahmad-S792 replied:
Just one sub-test regression due to extra `space`.

Ahmad-S792 replied:
Highlighted subtest specific below, so ignore this.

Ahmad-S792 commented on LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/nested-rules-001-expected.txt:
```diff
@@ -0,0 +1,3 @@
+
```
Progressed.

Ahmad-S792 commented on LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/margin-001-expected.txt:
```diff
@@ -0,0 +1,6 @@
+
+PASS There should be 3 @page rules.
+PASS Rule #0
```
Progressed.

Ahmad-S792 commented on LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-rules-001-expected.txt:
```diff
@@ -0,0 +1,10 @@
+
+PASS page-rules-001
+PASS @page , { } should be an invalid rule
+FAIL @page { } should be a valid rule assert_equals: serialization should be canonical expected "@page { }" but got "@page  { }"
```
This got regressed because of `extra` space.

darinadler replied:
Can we fix this please? No reason we have to land it broken. It should not be challenging.

Ahmad-S792 replied:
Yes - I am looking in background and will definitely try to not ship any regression.

Ahmad-S792 replied:
No regression anymore. :-)

Ahmad-S792 commented on LayoutTests/imported/w3c/web-platform-tests/css/cssom/cssom-pagerule-expected.txt:
```diff
@@ -1,14 +1,14 @@
-FAIL Cannot set selectorText to named page with pseudo, whitespace between assert_equals: expected "" but got "@page named"
-FAIL Cannot set selectorText to two pseudos, whitespace between assert_equals: expected "" but got "@page named"
-FAIL Cannot set selectorText to invalid pseudo page assert_equals: expected "" but got "@page named"
+PASS Page selector is initially the empty string
```
Two WPT progressions.

darinadler commented on Source/WebCore/css/CSSPageRule.cpp:
```diff
@@ -79,8 +79,8 @@ void CSSPageRule::setSelectorText(const String& selectorText)
-        return makeString(selectorText(), " { "_s, declarations, " }"_s);
-    return makeString(selectorText(), " { }"_s);
+        return makeString("@page "_s, selectorText(), " { "_s, declarations, " }"_s);
+    return makeString("@page "_s, selectorText(), " { }"_s);
```
```suggestion
    auto selector = selectorText();
    auto optionalSpace = selector.isEmpty() ? ""_s : " "_s;
    return makeString("@page"_s, optionalSpace, selector, " { }"_s);
```
Something like this but share the local variables with the other case too. Or could have 4 different calls to `makeString`, because the code is doing the same thing for `declarations` above.

Ahmad-S792 replied:
Thanks! I will test it locally now. :-)

Ahmad-S792 replied:
@darinadler - Thanks for help, pushed newer commit with `NO REGRESSION`, so this is just pure progression patch on WPT.

darinadler commented on Source/WebCore/css/CSSPageRule.cpp:
```diff
@@ -78,9 +78,11 @@ void CSSPageRule::setSelectorText(const String& selectorText)
     if (auto declarations = m_pageRule->properties().asText(); !declarations.isEmpty())
-        return makeString(selectorText(), " { "_s, declarations, " }"_s);
-    return makeString(selectorText(), " { }"_s);
+        return makeString("@page "_s, selector, " { "_s, declarations, " }"_s);
```
```suggestion
        return makeString("@page"_s, optionalSpace, selector, " { "_s, declarations, " }"_s);
```

Ahmad-S792 replied:
It was not progressing the failing test, so I didn't done the change here. Although now I have done, let's see from EWS, if it do any progression or regression. :-)

darinadler replied:
If there is no progression then that tells us we have insufficient test coverage! Almost certainly not that the change is unneeded.

Ahmad-S792 replied:
When I did earlier without `space`, it did regress few things. So I think we do have test coverage but I was keeping it `@page `, so it was not reflecting. I remember one of my EWS showing failure when I removed space from first return.

darinadler replied:
The two consecutive spaces would be a bug. We should do it correctly and make sure the tests expect the correct behavior.

annevk replied:
@Ahmad-S792 is this tested now?

Ahmad-S792 replied:
@annevk - thanks for your guidance on Slack 1-1, I have extended - `css/cssom/cssom-pagerule.html` to test `cssText` path as well. Appreciate if you can review. :-)

annevk commented on Source/WebCore/css/CSSPageRule.cpp:
```diff
@@ -57,10 +57,10 @@ String CSSPageRule::selectorText() const
-        if (!pageSpecification.isEmpty() && pageSpecification != starAtom())
-            return makeString("@page "_s, pageSpecification);
+        if (!pageSpecification.isEmpty())
+            return pageSpecification;
```
Given the `return ""_s` below doesn't this simplify to `return selector->selectorText()`?

Ahmad-S792 replied:
Update - noted, we need to return initially `emptyString()` so I kept it `""_s`.

annevk commented on Source/WebCore/css/CSSPageRule.cpp:
```diff
@@ -56,11 +56,10 @@ CSSStyleDeclaration& CSSPageRule::style()
-        String pageSpecification = selector->selectorText();
-        if (!pageSpecification.isEmpty() && pageSpecification != starAtom())
-            return makeString("@page "_s, pageSpecification);
+        if (!selector->selectorText().isEmpty())
+            return selector->selectorText();
```
```suggestion
        return selector->selectorText();
```
This is equivalent, no?

darinadler replied:
It's not necessarily the same if `selectorText()` can return the null string. The code as written will return the non-null empty string if `selectorText()` returns the null string.

annevk commented on Source/WebCore/css/CSSPageRule.cpp:
```diff
@@ -56,11 +56,10 @@ CSSStyleDeclaration& CSSPageRule::style()
+            return selector->selectorText();
     }
-    return "@page"_s;
+    return ""_s;
```
This should probably use `emptyString()`?

darinadler replied:
At one time that was slightly more efficient. Not sure if that's still the case.

darinadler reviewed (approved):
I don't think either of the last two remaining comments are a must. This can land now.

Ahmad-S792 committed:
Align `selectorText` to not serialize `@page` and not be nullable  as per Web Specification

https://bugs.webkit.org/show_bug.cgi?id=271432
rdar://125588212

Reviewed by Darin Adler.

This patch aligns WebKit with Web Specification [1] & [2]:

[1] https://drafts.csswg.org/cssom/#dom-csspagerule-selectortext
[2] https://drafts.csswg.org/cssom/#csspagerule

As per Web Specification, it should not append `@page`, so now
it serialize correctly. Similarly, it should not be nullable.

This patch also syncs WPT tests (not printing specific) below:

Upstream commit: https://github.com/web-platform-tests/wpt/commit/c72ee0c082d350b8b44f7a6bac1f028752b13527

* Source/WebCore/css/CSSPageRule.cpp:
(WebCore::CSSPageRule::selectorText const):
(WebCore::CSSPageRule::cssText const):
* Source/WebCore/css/CSSPageRule.idl:

> Rebaselines (Local Tests)
* LayoutTests/http/wpt/cssom/CSSPageRule.html: Updated
* LayoutTests/fast/dom/css-element-attribute-js-null-expected.txt: Synced from Blink Source
* LayoutTests/fast/dom/css-element-attribute-js-null.html: Synced from Blink Source

> WPT Tests Update & Sync:
* LayoutTests/imported/w3c/resources/import-expectations.json:
* * LayoutTests/imported/w3c/web-platform-tests/css/cssom/cssom-pagerule.html: Extended for `cssText` cases as well
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/cssom-pagerule-expected.txt: Rebaselined
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/margin-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/margin-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/margin-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/margin-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/margin-003-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/margin-003.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/page-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/page-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/page-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/page-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/cssom/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/inheritance-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/inheritance.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/META.yml:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-orientation.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-orientation.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-000-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-000.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-003-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-003.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-004-expected.txt::
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/page-rule-declarations-004.html::
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/margin-rules-001-expected.txt::
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/margin-rules-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/margin-rules-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/margin-rules-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/nested-rules-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/nested-rules-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-invalid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-invalid.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-orientation-computed.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-orientation-computed.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-orientation-invalid.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-orientation-invalid.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-rules-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-rules-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/page-valid.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/parsing/WEB_FEATURES.yml:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/trailing-declaration-crash.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/css-page/WEB_FEATURES.yml:

Canonical link: https://commits.webkit.org/289428@main
